cdap-api/src/main/java/co/cask/cdap/api/metadata/plugin/EntityMetadataReaderContext.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package co.cask.cdap.api.metadata.plugin;
```

Comment:
If it is API for plugin, it shouldn't be inside `cdap-api` module. It should belongs to the data-pipeline app, hence under the `cdap-etl-api`.